src/test/java/bio/terra/workspace/service/workspace/flight/CreateGoogleContextFlightTest.java

```diff
@@ -0,0 +1,136 @@
+package bio.terra.workspace.service.workspace.flight;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import bio.terra.cloudres.google.cloudresourcemanager.CloudResourceManagerCow;
+import bio.terra.cloudres.google.serviceusage.ServiceUsageCow;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.FlightState;
+import bio.terra.stairway.FlightStatus;
+import bio.terra.stairway.Stairway;
+import bio.terra.workspace.common.BaseIntegrationTest;
+import bio.terra.workspace.common.StairwayTestUtils;
+import bio.terra.workspace.db.WorkspaceDao;
+import bio.terra.workspace.service.job.JobService;
+import bio.terra.workspace.service.workspace.WorkspaceCloudContext;
+import com.google.api.services.cloudresourcemanager.model.Project;
+import com.google.api.services.serviceusage.v1.model.GoogleApiServiceusageV1Service;
+import com.google.api.services.serviceusage.v1.model.ListServicesResponse;
+import java.time.Duration;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class CreateGoogleContextFlightTest extends BaseIntegrationTest {
+  /** How often to poll Stairway for flight completion. */
+  private static final Duration STAIRWAY_POLL_INTERVAL = Duration.ofSeconds(5);
+  /** How long to wait for a Stairway flight to complete before timing out the test. */
+  private static final Duration STAIRWAY_FLIGHT_TIMEOUT = Duration.ofMinutes(5);
+
+  @Autowired WorkspaceDao workspaceDao;
+  @Autowired CloudResourceManagerCow resourceManager;
+  @Autowired ServiceUsageCow serviceUsage;
+  @Autowired private JobService jobService;
+
+  @Test
+  public void successCreatesProjectAndContext() throws Exception {
+    UUID workspaceId = createWorkspace();
+    assertEquals(WorkspaceCloudContext.none(), workspaceDao.getCloudContext(workspaceId));
+
+    FlightMap inputParameters = new FlightMap();
+    inputParameters.put(WorkspaceFlightMapKeys.WORKSPACE_ID, workspaceId);
+
+    Stairway stairway = jobService.getStairway();
+    String flightId = stairway.createFlightId();
+    stairway.submit(flightId, CreateGoogleContextFlight.class, inputParameters);
+    FlightState flightState =
+        StairwayTestUtils.pollUntilComplete(
+            flightId, stairway, STAIRWAY_POLL_INTERVAL, STAIRWAY_FLIGHT_TIMEOUT);
+    assertEquals(FlightStatus.SUCCESS, flightState.getFlightStatus());
+
+    String projectId =
+        flightState
+            .getResultMap()
+            .get()
+            .get(WorkspaceFlightMapKeys.GOOGLE_PROJECT_ID, String.class);
+    assertEquals(
+        WorkspaceCloudContext.createGoogleContext(projectId),
+        workspaceDao.getCloudContext(workspaceId));
+    Project project = resourceManager.projects().get(projectId).execute();
```

Comment:
Do we need to mark this test or do anything for Janitor, or will it just handle cleaning this up?

Reply:
As long as CRL is configured to talk to the Janitor, we're good. From previous PR, we have CRL configured to do so with testing profiles. So, as structured, these tests work.
For preview environments, or other non-persistent environments, we'll need to think more about how to start up WM with CRL in testing mode so that it does talk to Janitor.